community/kernel/src/main/java/org/neo4j/kernel/impl/factory/DataSourceModule.java

```diff
@@ -331,7 +333,16 @@ private Procedures setupProcedures( PlatformModule platform, CoreAPIAvailability
         ProcedureGDSFactory gdsFactory = new ProcedureGDSFactory( platform.config, platform.storeDir, platform.dependencies, storeId, this.queryExecutor, coreAPIAvailabilityGuard, platform.urlAccessRule );
         procedures.registerComponent( GraphDatabaseService.class, gdsFactory::apply );
 
+        // Below components are not public API, but are made available for internal
+        // procedures to call, and to provide temporary workarounds for the following
+        // patterns:
+        //  - Batch-transaction imports (GDAPI, needs to be real and passed to background processing threads)
+        //  - Group-transaction writes (same pattern as above, but rather than splitting large transactions,
+        //                              combine lots of small ones)
+        //  - Bleeding-edge performance (Statement, to bypass overhead of working with Core API)
         procedures.registerComponent( DependencyResolver.class, (ctx) -> platform.dependencies );
+        procedures.registerComponent( Statement.class, ( ctx ) -> ctx.get( KERNEL_TRANSACTION ).acquireStatement() );
```

Comment:
I'm not sure we should inject a statement at all. acquiring one up front means we never can release the current statement and get another one withing the current transaction (for example Cypher does this to free locks in one case I believe). Let's rather inject the whole kernel transaction for "hidden" statement access instead and leave calling acquire and relase to the built-in procedure.